### DISCUSSION TITLE
Deploying a MonoGame Example to Github Pages (or Similar)

### DISCUSSION THREAD

**Discussion by egocarib** [MonoGame/MonoGame#8454] [Content]
I'm attempting to use [kniEngine ](https://github.com/kniEngine/kni) by @nkast to deploy a very simple MonoGame game to GitHub Pages. I was attempting to follow this tutorial: [MonoGame on the Web: No Really!](https://darkgenesis.zenithmoon.com/monogame-on-the-web-No-really!.html)

I am hitting an issue where it can't find MGCB.exe when going through the dotnet publish GitHub action ([see failure here](https://github.com/egocarib/DadGames/actions/runs/10498164654/job/29082553514#step:5:12)). The dotnet publish command works fine when I run it in my local PC directory.

Hoping someone can help steer me in the right direction for how to resolve this, or otherwise point me to a different tutorial/resource that can help me get a simple MonoGame published to the web (with or without kniEngine). Doesn't necessarily have to be on GitHub Pages, but I thought that would be the easiest place to start.

Thanks!

**Comment by nkast** [comment]
You need to run on a windows machine,   and install net8. 

see 'runs-on:'  and 'Setup .NET' here : 
 https://github.com/kniEngine/kni/blob/main/.github/workflows/dotnet.yml
 
 for the record, I never tried to build a project with the pipeline builder before. 
 That action is what I  use to build the libraries.

**Reply by egocarib** [reply]
Thank you, that was indeed the problem. The game is working and live on GitHub Pages now! :)

**Comment by nkast** [comment]
That's great!

I found the games url , 
https://egocarib.github.io/DadGames/ 

Good to know that this 'just works' out of the box.
I think I will, automate some of my projects too...

**Comment by SimonDarksideJ** [comment]
Nice tutorial find @egocarib , make sure to thank the author :P 

P.S. Don't forget to flip the texture when the cat walks left `Scale = new Vector2(-1,0)`

**Reply by egocarib** [reply]
Hah! Thank you @SimonDarksideJ! 😸
